app/src/main/java/org/schabi/newpipe/player/PlayerIntentType.kt

```diff
@@ -0,0 +1,26 @@
+package org.schabi.newpipe.player
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+// We model this as an enum class plus one struct for each enum value
+// so we can consume it from Java properly. After converting to Kotlin,
+// we could switch to a sealed enum class & a proper Kotlin `when` match.
+
+@Parcelize
+enum class PlayerIntentType : Parcelable {
+    Enqueue,
+    EnqueueNext,
+    TimestampChange,
+    AllOthers
+}
```

Comment:
Enums are serializable, so they can be passed/retrieved from intents/bundles as serializable objects.

Reply:
what’s the difference? I don’t see any.

Reply:
```suggestion
// We model this as an enum class plus one struct for each enum value
// so we can consume it from Java properly. After converting to Kotlin,
// we could switch to a sealed enum class & a proper Kotlin `when` match.

enum class PlayerIntentType {
    Enqueue,
    EnqueueNext,
    TimestampChange,
    AllOthers
}
```